File: ball.js
function ball() {
    this._inSide = "";
    this.__div = document.createElement("div");
    js.addClass(this.__div, "ball");
};

js.extend(ball, objectEx);

ball.prototype.getInSide = function(){
    return this._inSide;
};

ball.prototype.getDiv = function(){
    return this.__div;
};

ball.prototype.setInSide = function(v){
    this._inSide = v;
};

File: ball.cs
using SharpKit.Html;
using SharpKit.JavaScript;
using SharpKitWebApp.core;

namespace SharpKitWebApp
{
    [JsType(JsMode.Prototype, Filename = "gen/ball.js")]
    public class ball
    {
        private readonly HtmlElement __div;
        private ESlide _inSide;

        public ball()
        {
            _inSide = ESlide.none;
            __div = HtmlContext.document.createElement("div") as HtmlElement;
            js.addClass(__div, "ball");
        }


        public ESlide getInSide()
        {
            return _inSide;
        }

        public void setInSide(ESlide str)
        {
            _inSide = str;
        }

        public HtmlElement getDiv()
        {
            return __div;
        }
    }
}